internal/extproc/server.go

```diff
@@ -178,3 +192,56 @@ func (s *Server[P]) Check(context.Context, *grpc_health_v1.HealthCheckRequest) (
 func (s *Server[P]) Watch(*grpc_health_v1.HealthCheckRequest, grpc_health_v1.Health_WatchServer) error {
 	return status.Error(codes.Unimplemented, "Watch is not implemented")
 }
+
+// filterSensitiveHeaders filters out sensitive headers from the provided HeaderMap.
+// Specifically, it redacts the value of the "authorization" header and logs this action.
+// The function returns a new HeaderMap with the filtered headers.
+func filterSensitiveHeaders(headers *corev3.HeaderMap, logger *slog.Logger, sensitiveKeys []string) *corev3.HeaderMap {
+	if headers == nil {
+		logger.Debug("received nil HeaderMap, returning empty HeaderMap")
+		return &corev3.HeaderMap{}
+	}
+	filteredHeaders := &corev3.HeaderMap{}
+	for _, header := range headers.Headers {
+		if slices.Contains(sensitiveKeys, header.GetKey()) {
```

Comment:
if we want to be extra cautious (which I believe is what we want), can we use the strings.ToLower(maybe the name is different ;) to ensure the case-insensitivity